### PR TITLE
Rehab the plugin

### PR DESCRIPTION
**Pull request by scorphus**
/cc @brunobbbs and @pantuza

### CODE REVIEW

**Comment by Lukas0907** [comment]
Hi,

I think this can be solved with pyenv more easily. Running "pyenv init -" from fish already prints the following stub:

```sh
set -gx PATH '/home/lukas/.pyenv/shims' $PATH
set -gx PYENV_SHELL fish
source '/home/lukas/.pyenv/libexec/../completions/pyenv.fish'
command pyenv rehash 2>/dev/null
function pyenv
  set command $argv[1]
  set -e argv[1]

  switch "$command"
  case activate deactivate rehash shell
    source (pyenv "sh-$command" $argv|psub)
  case '*'
    command pyenv "$command" $argv
  end
end
```

So shipping the pyenv function and the completions in this plugin is not necessary.

To make plugin-pyenv more useful, I think it would be enough to source the output of pyenv init - any maybe the output of the virtualenv-init subcommand (see also #1).

**Review by brunobbbs — APPROVED** [review]

**Comment by BarbzYHOOL** [comment]
I just tested this and it works fine. Prior to the PR, it didn't load pyenv unless I added some extra vars to the config

**Comment by BarbzYHOOL** [comment]
It works when I install the plugin the first time (with fundle install) but then after I open a new terminal, pyenv command never exists so basically the $PATH is not set

It's the first line of the init.fish that fails

I even tried with `set PYENV_ROOT $HOME/.pyenv` but it doesn't set it o_0

**Comment by scorphus** [comment]
@Lukas0907 we should try our best to keep the plugin as lazy as possible. Imagine how long a fish user would need to wait for the command prompt if every single installed plugin would execute one or more commands on `init`.

@BarbzYHOOL thanks for chiming in and calling for action here. Why do you need the `PYENV_ROOT` variable for?

**Comment by BarbzYHOOL** [comment]
because the plugin is not working at all for me, nothing in init.fish is read o_0

And look at the latest file it's here https://github.com/oh-my-fish/plugin-pyenv/pull/5/files

**Comment by scorphus** [comment]
@BarbzYHOOL do you know how to test that PR? In the meantime, I'll make sure it works on some Docker images. The I'll merge that PR.

**Comment by BarbzYHOOL** [comment]
I tested the PR and even merged it in my own fork and I'm using it at the moment (with the bugs I reported above)

The only difference is that I use "fundle" instead of fisher (I opened an issue in fundle but I checked other plugins I use, and they have the same exact init.fish and they work fine, in top of that if I copy paste the content of plugin-pyenv/init.fish into another-plugin/init.fish it works fine o_0)

**Comment by scorphus** [comment]
@BarbzYHOOL I've just tested the `rehab` branch in both Oh My Fish and Fundle. Both fresh installs on two separate [Fish Shell 2.7.1 containers](https://hub.docker.com/r/ohmyfish/fish/tags/). I can share both with you, if you'd like.

Regarding Fundle, I just followed instructions in https://github.com/danhper/fundle#installation and https://github.com/danhper/fundle#usage.

Can you test it that way too?

**Comment by BarbzYHOOL** [comment]
All my other plugins work fine, I use fundle for months

and I checked, my plugin-pyenv repo (in fundle/) has the right commits:

```
4602a38 2018-11-11 Barbz (HEAD, origin/master, origin/HEAD) Merge pull request #2 from BarbzYHOOL/rehab 
287d8ef 2018-11-11 Barbz Merge branch 'master' into rehab 
887e2ae 2018-11-11 Barbz Add missing commands from official pyenv install 
2e00b6c 2018-01-31 Pablo Santiago Blum de Aguiar (origin/rehab) Improve README.md a bit 
53dad28 2017-05-26 Gustavo Pantuza Readme: Create base documentation of how to use plugin 

```

So it's working for you on fundle ??

Did you exit your terminal, and open another? because as I said, the first installation works, then after it never works again

**Comment by scorphus** [comment]
Yep. I made it work. Then made it stop working by commenting contents of `init.fish`. Then make it work again by _uncommenting_ them.

So, as soon as I can, I'll share with you the Docker image I used and you can check how it goes for you.

**Comment by BarbzYHOOL** [comment]
I don't use docker so don't send me the docker image (but thank you for proposition). hmmm there must be something wrong with me but I have no idea what :'(
I haven't made weird stuff like symlink etc

I have no idea what to do :(

**Comment by scorphus** [comment]
Let's see if I can help you. What Fish version are you running? Can you share the content of `config.fish` and also the content of `plugin-pyenv`'s `init.fish`?

**Comment by BarbzYHOOL** [comment]
omg  I use two config file, the normal one and one in /etc/fish. I just posted and saw that there is no `fundle init` in the first config file.

so it only affected my plugin-pyenv because I have only one plugin in that conf.fish

So it's solved thanks to you @scorphus haha

File: init.fish
Comment (scorphus) [inline init.fish]: I'll change this line as `^` operator is deprecated.

**Comment by derekstavis** [comment]
@scorphus any chance this is still good to merged?

**Comment by scorphus** [comment]
Oh, man. I keep forgetting this PR. I'm still using this branch with a small unstaged change on top. So far it's been working fine. Did you by any chance manage to test it?

**Comment by BarbzYHOOL** [comment]
i think I tested it from the msg I put above

**Comment by scorphus** [comment]
So, @derekstavis, once it has your approval (or approval from any of the core devs) as usual, I'll pull `master`.

**Review by derekstavis — APPROVED** [review]

**Comment by derekstavis** [comment]
Done! Thanks for following up here!

**Comment by scorphus** [comment]
Thanks for contributing/testing/reviewing, @BarbzYHOOL, @brunobbbs, @pantuza, and @derekstavis! 🐠🐚

**Pull request merged**
